File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n = len(nums)
        curr_sum = nums[0]
        max_sum = nums[0]
        for i in range(1, n):
            curr_sum += nums[i]
            curr_sum = max(curr_sum, nums[i])
            max_sum = max(max_sum, curr_sum)
        return max_sum
 
            

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int curr_sum = nums[0];
        int max_sum = nums[0];
        int n = nums.length;
        for(int i = 1; i < n; i++){
            curr_sum += nums[i];
            curr_sum = Math.max(curr_sum, nums[i]);
            max_sum = Math.max(curr_sum, max_sum);
        }
        return max_sum;
    }
}